Makefile

```diff
@@ -72,15 +88,22 @@ build-docker:
 build-docker-git:
 	git archive HEAD | docker build $(DOCKER_ARGS) -t $(DOCKER_TAG) -
 
-Gopkg.lock: Gopkg.toml
-	mkdir -p .go/src/github.com/zededa && ln -s ../../../.. .go/src/github.com/zededa/go-provision || :
-	GOPATH=$(CURDIR)/.go go get github.com/golang/dep/cmd/dep
-	rm vendor
-	mv src/vendor vendor
-	cd .go/src/github.com/zededa/go-provision ; PATH=$(CURDIR)/.go/bin:$$PATH GOPATH=$(CURDIR)/.go \
-          dep ensure -update $(GODEP_NAME)
-	mv vendor src/vendor
-	ln -s src/vendor vendor	
+export BUILD_CONTAINER
+eve-build-$(USER):
+	@echo "$$BUILD_CONTAINER" | docker build -t $@ - >/dev/null
+
+shell: eve-build-$(USER)
+	@docker run -it --rm -u $(USER) -w /home/$(USER) \
+	  -v $(CURDIR)/.go:/go -v $(CURDIR):/go/src/$(GOMODULE) -v $${HOME}:/home/$(USER) \
+	$< $(SHELL_CMD)
```

Comment:
So the purpose of `make shell` is to launch into a convenient environment where I can run any `go` commands I want?

A few suggestions:

1. We should have `make` installed if we do this via `apk add make`
2. Why not change directory to `/go/src/$(GOMODULE)`, so you are ready to work where you are?
3. We should change the `build` target to do the build via docker, at least as an option, so I can just run `make build` and get a dockerized build from my local environment (without having to drop into an interactive docker-based shell).

I will put a suggestion in lower comments.

Reply:
All good points. Do you think I can push it as-is and you can add all these good ideas later?